test/src/tests-batchUploader.ts

```diff
@@ -88,13 +204,15 @@ describe('batch uploader', () => {
                 uploader.queueEvent(({} as unknown) as SDKEvent);
 
                 expect(uploader.pendingEvents).to.eql([]);
-                expect(uploader.pendingUploads).to.eql([]);
+                // TODO: Should we remove because of Vault?
+                // expect(uploader.pendingUploads).to.eql([]);
```

Comment:
i think the answer to this depends on the flow chart you make. we need to be really clear on when things remain in memory vs when they are cleared.  it's a layer of complexity that needs to be diagrammed